### ISSUE TITLE
Step 2: Sidenav Design - Accessibility review

### ISSUE THREAD

**Issue by caw310** [department-of-veterans-affairs/vets-design-system-documentation#4340]
## Issue Description
We need to conduct a design accessibility review of the sidenav component. 

## Tasks
- [x] Accessibility review of the design of the component
- [ ] If mobile component validate on mobile devices the component accessibility behaviors match expectations
     - [ ] Screen readers
     - [ ] Text size changes
- [x] Provide input to the designer with accessibility concerns 
- [ ] If necessary, write up a ticket with those issues 

## Acceptance Criteria
- [x] Design of component has been reviewed for accessibility issues
- [ ] Any accessibility issues have been put in a new ticket if needed

**Comment by jeana-adhoc** [comment]
Findings:

**Have we documented….**
1. **The main purpose of this experience, interaction, component?** [From product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/design-system-forms-library/products/components/va-sidenav/product-outline.md) “The Sidenav component provides a consistent, accessible, and responsive side navigation panel for web applications. It is designed to help users quickly navigate between major sections or features within a page or application. The Sidenav is intended for use in layouts where persistent or easily accessible navigation is essential, such as dashboards, admin panels, and complex content structures” 
2. **Is there a USWDS equivalent? Do we have sufficient reason/research to deviate from it?** There is a [USWDS equivalent](https://designsystem.digital.gov/components/side-navigation/) [ASSUMPTION] We are deviating from it because side nav already exists on [VA.gov](http://va.gov/) and we need to merge supporting the current structure while building a component that will be as close to uswds as possible?
    - USWDS doesn’t use accordions, etc
    - USWDS doesn’t have “categories”

**Accessibility questions**
1. **The Nav region needs an accessible name.** We should define it for the component.  `aria-label=””` Suggestions:
    - Pages related to [this section] or [h2]
    - Pages in this section (strict definition - links never leave the section)
    - Might be research here that shows this may be confusing
    - [h2] Pages
    - Side navigation for [h2] or Secondary or Section ? 
    - Questions - will there always be an H2? Or do we need a fall back? Or should it be generic. At the end of the day we want this decided/defined in the component.
2. **Mobile/Small screen view:** To make sure we’ve accounted for Focus and content reflow/zoom issues, we need to have an understanding of the mobile behavior. Will it be sticky? A drawer? An accordion with everything else in it (aka a nested accordion?) Accordions preferred over a drawer ([example](https://www.va.gov/chicago-health-care/locations/lakeside-va-clinic/))
3. **The “Current Link”/Active Page** There are various ways we can announce it & code it & display it..
    - Preferred method: <a href=”#main”>Health benefits</a>... this way it won’t refresh the page and still navigates the user to the region like a skip link
    - A screen reader will announce this as “Health benefits, same page link”. No need for extra aria or hidden sr text. (Will try to have a codepen link in time for the meeting)
    - Since it is indeed a link, we should visually style it as a link, while using a  different visual treatment for the current page link, but maintaining the semantic integrity of it being a link
4.  All links in a side nav
    - Could we introduce border-bottom or underline to these links as well?  (But we don’t do that in any other menu structure)
    - Could we create parity with other navigation links like what we’re doing in tabs, where the current item has a left blue border? The concern: Links should look like links. The ‘current’ item is still a link.
    - Tabs <br /><img width="212" alt="Image" src="https://github.com/user-attachments/assets/4f19c3da-39f2-4e76-b7af-cab525faf019" />
     - USWDS <br /><img width="180" alt="Image" src="https://github.com/user-attachments/assets/33b0884d-a284-4a99-b08b-7b461766e5e3" />
    - Proposed <br /><img width="196" alt="Image" src="https://github.com/user-attachments/assets/f99d63c1-333b-412d-b7aa-f92d06d108c3" />

**Comment by jeana-adhoc** [comment]
After DSC on June 26, the decisions were made...

- Go forward with v1 design.
- Category name that is not interactive is bold, all caps, black text.
- No more than 5 links within a category

No additional items need to be addressed in accessibility for v1.  We'll want to review future iterations (v2.. v3.. ) of the component
